### ISSUE TITLE
Disparity between checks against test "command" and [[ "command" ]] 

### ISSUE THREAD

**Issue by jamesisin** [koalaman/shellcheck#2641]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1009, SC1073, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

# The issue is that the version of test using [[]] fails in ShellCheck but the version using "test" passes.  
# If they are both legitimate... 
# fails:  
# if [[ "$( find ${const_killFilePath} -mmin +"${const_threasholdCrit}" )" && ! grep -Fq "KILL_MODE=1" "${const_killFilePath}" ]] ; then 
# passes:  
# elif test "$( find ${const_killFilePath} -mmin +"${const_threasholdWarn}" )" && ! grep -Fq "KILL_MODE=1" "${const_killFilePath}" ; then  
# OR 
# If the [[]] version cannot be used for testing command output (as written above), then SC should indicate that is the case and offer an explanation 

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /usr/bin/env bash 
# Title   :  KillModeAgeCheck.sh 
# Parent  :  n/a 
# Author  :  JamesIsIn 20221206  Do something nice today.  

# Purpose :  Check length of time Kill Mode has been disabled and alert after X time.  
# https://esentire.atlassian.net/wiki/spaces/MON/pages/240451638/esNETWORK-00036+KILL+MODE 
# https://esentire.atlassian.net/wiki/spaces/~378412971/pages/1570112466/Some+random+notes+on+Kill+Mode 
# 

## 

############### 
#  Variables  # 

declare killModeCheck 
	killModeCheck="0" 
declare solarwinds_initiation 
	solarwinds_initiation="1" 
readonly const_killFilePath="redacted" 
readonly const_threasholdWarn="1" # for debugging times can be dropped to single-digit minutes 
readonly const_threasholdCrit="2" # for debugging times can be dropped to single-digit minutes 

# # debugging 
# 
# # 

## 

############### 
#  Functions  # 

function func_checkKillMode() { 
	# if file does not exist default is enabled 
	if [[ "${const_killFilePath}" ]] ; then 
		# no file = Kill Mode enabled = ok 
		# KILL_MODE=1 # set to 1 to allow spoofed packets (ie kills), anything else will not allow kills 
		# if test $( find /home/jcook/test.sh -mmin +3015120 ) ; then echo "yes" ; fi 
		# mmin = mod time ; cmin = change ; amin = access time -- in minutes 
		if [[ "$( find ${const_killFilePath} -mmin +"${const_threasholdCrit}" )" && ! grep -Fq "KILL_MODE=1" "${const_killFilePath}" ]] ; then 
			# check crit 
			killModeCheck="2" 
		elif test "$( find ${const_killFilePath} -mmin +"${const_threasholdWarn}" )" && ! grep -Fq "KILL_MODE=1" "${const_killFilePath}" ; then 
			# check warn 
			killModeCheck="1" 
		fi 
	fi 
} 

function func_output() { 
	# if --solarwinds SW output otherwise human output 
	if [ ${solarwinds_initiation} = 0 ] ; then 
		# if SW then output special key/value pair 
		# values are 0=ok 1=warn and 2=crit 
		if [ ${killModeCheck} = 0 ] ; then 
			printf '%s\n' "OK:  Kill Mode is enabled.  " 
			return 0 
		elif [ ${killModeCheck} = 1 ] ; then 
			printf '%s\n' "WARN:  Kill Mode has been disabled for more than ${const_threasholdWarn} minutes.  " 
			return 1 
		else 
			printf '%s\n' "CRIT:  Kill Mode has been disabled for more than ${const_threasholdCrit} minutes.  " 
			return 2 
		fi 
	else 
		# human output 
		if [ ${killModeCheck} = 0 ] ; then 
			printf '%s\n' "OK:  Kill Mode is enabled.  " 
			return 0 
		elif [ ${killModeCheck} = 1 ] ; then 
			printf '%s\n' "WARN:  Kill Mode has been disabled for more than ${const_threasholdWarn} minutes.  " 
			return 1 
		else 
			printf '%s\n' "CRIT:  Kill Mode has been disabled for more than ${const_threasholdCrit} minutes.  " 
			return 2 
		fi 
	fi 
} 

function func_parseArguments() { 
	# 
	case "$1" in 
		# use a case statement to differentiate between user-induced and script-induced versions if required 
		--solarwinds) 
			# set variable and avoid interaction; if variables are missing throw an error 
			solarwinds_initiation="0" 
			shift 1 ;; 
	esac 
} 

function main() { 
	# 
	func_parseArguments "$@" 
	func_checkKillMode 
	func_output 
} 

## 

########## 
#  Main  # 

main "$@" 

exit $? 

## 

```

#### Here's what shellcheck currently says:
```
Line 40	SC1009: The mentioned syntax error was in this if expression.
Line 40	SC1073: Couldn't parse this test expression. Fix to allow more checks.
Line 40	SC1072: Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by aeiplatform** [comment]
Why do you use test for checking exit codes?

**Comment by aeiplatform** [comment]
`if command && ! other; then`

**Comment by jamesisin** [comment]
Whether or not this is ideal code ShellCheck should be able to parse both test and [[]].  Currently parsing this line halts inside the [[]] version between the two quoted expressions after grep where ShellCheck mistakenly takes the test command to end.  This error (by ShellCheck) does not occur in the test version.  ShellCheck should be able to parse both versions.  

It would seem useful for ShellCheck to suggest not using test in these scenarios, sure.  It certainly (and helpfully) does offer this kind of advice elsewhere.  But this would be secondary to getting ShellCheck to parse the line at all.
